OPHD/States/MapViewState.h

```diff
@@ -230,7 +230,7 @@ class MapViewState : public Wrapper
 	int mFood{ 0 };
 
 	// POOLS
-	const StorableResources mPlayerResources; /**< Player's current refined resources. */
+	StorableResources mPlayerResources; /**< Player's current refined resources. */
```

Comment:
This should be const. mPlayerResources is used to provide feedback to the player of what they have in storage, it is not a pool to be drawn from. By making it const it prevents accidentally drawing from that pool.

This could be worked around by tallying up the players's current resources in a different way, I just use the StorableResources class as a sort of shortcut.

Reply:
Do you mean it should be a const reference to data stored elsewhere? Because that's not how it's used here. It is actually being written to.

Perhaps we could change how resources are recalculated and revisit this. Currently though, there is no feasible way to make that field `const` unless we move responsibility to recalculating that value outside of the class and make it a `const StorableResources&` field.

Reply:
I know it's being written to. That was the intent, to write only in that part of the code so that we're not counting resources stored in each structure every frame to get a tally.

This is literally a shortcut. This could be done better by just having a separate array in `MapViewState`, something like `std::array<int, 4> mResourceCount` and then just writing to that each update. That's probably the 'correct' way to do it but I'm lazy.